ceph_iscsi_config/gateway.py

```diff
@@ -412,14 +412,21 @@ def map_luns(self, config):
         target_config = config.config["targets"][self.iqn]
         target_stg_object = [stg_object for stg_object in lio_root.storage_objects
                              if stg_object.name in target_config['disks']]
+        
+        # a dict, key with tpg and storage object name
+        tpg_stg_object_dict = {}
+        for tpg in self.tpg_list:
 
+             for l in tpg.luns:
+                tpg_stg_object_dict[str(tpg.tag) + '-' + l.storage_object.name] = l.storage_object.name
+        
         # process each storage object added to the gateway, and map to the tpg
         for stg_object in target_stg_object:
 
             for tpg in self.tpg_list:
                 self.logger.debug("processing tpg{}".format(tpg.tag))
 
-                if not self.lun_mapped(tpg, stg_object):
+                if not tpg_stg_object_dict.has_key(str(tpg.tag) + "-" + stg_object.name):
```

Comment:
Did you see my comment:

https://github.com/ceph/ceph-iscsi-config/pull/100#discussion_r250003141

LUN() does a exists check for us, so there is no need to create the dict. Did my suggestion in that comment not perform as well as the dict based approach above?

Reply:
yeah, i see. but when catch a exception, configuration will be cleaned and the program exit.

Reply:
![image](https://user-images.githubusercontent.com/3916975/51718193-29754f00-207f-11e9-90da-522ea4935552.png)

Reply:
It's not clear to me how you hit that. Won't the code:

```
        except RTSLibError as err:
            if "already exists in configFS" not in str(err):
                self.logger.error("LUN mapping failed: {}".format(err))
                self.error = True
                self.error_msg = str(err)
                return

           #Already created. Ignore and loop to the next tpg.
           continue
```

Catch the exception so it does not get escalated up to exception_handler()?

Reply:
use this code, rbd-target-gw.log with no error 
![image](https://user-images.githubusercontent.com/3916975/51811812-c509f800-22e9-11e9-9dcf-6ce181780766.png)